src/middlewares/auth.js

```diff
@@ -0,0 +1,27 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const authentication = (req, res, next) => {
+  const token = req.headers.authorization || req.headers['x-access-token'];
+  if (token) {
```

Comment:
Token will contain something like this: _Bearer token_string_ . E.g: _Bearer d$hfhhjdnjijdh_ and not _d$hfhhjdnjijdh_
In your case, your token verification is looking like:
`Jwt.verify(Bearer d$hfhhjdnjijdh,....` instead of `Jwt.verify(d$hfhhjdnjijdh,....`
I think it is better to extract the token from the string (Bearer token) using the split method. E.g:

`const bearerHeader = req.headers.authorization;

	if (bearerHeader ) {
		const bearer = bearerHeader.split(' ');
		// get token from array after split
		const token = bearer[1];
		jwt.verify(token, 'secretkey', (err, decoded) => {`

And I did not see where you called the **next** function
And I thought we should be using async await

Reply:
Thank you for the feedback. Please review closely you will see the **next** function and there are no callbacks nor promises in my code, so can you please specify the need for **async/await**. I will implement splitting the bearer token.

Reply:
My bad. I saw the next. **async/await** was specified y the TTL to be used in place of promises/callbacks.
You code can be re-written as :
```
const authorization = async (req, res, next) => {
........
try {
const decode = await JWT.verify(token, process.env.TOKEN_SECRET);
req.userData =decode;
return next()
// etc
}
catch(err) {
// do something with error
}
```

you can reach out to him to know the current position on the use of async/await